### PR TITLE
Improve plugin related names

### PR DESCRIPTION
**Pull request by mkoistinen**
This is the continuing saga of Django 1.9 vs. cmsplugin_ptr. The changes in this PR were initially contributed by @yakky's #5032.

### CODE REVIEW

**Comment by czpython** [comment]
why cms/tests/test_frontend.py → cms/tests/DISABLED_test_frontend.py ?

File: cms/models/pluginmodel.py [hunk 1]
```diff
@@ -48,6 +48,20 @@ def __new__(cls, name, bases, attrs):
...
+        # related descriptor to avoid name clashes for plugins with the same
+        # fieldnames. Skip this check if the parent-child field has been
+        # explicitly defined in the plugin subclass
+        #
+        # NOTE: We only do concrete classes, skipping the abstract ones
+        if 'cmsplugin_ptr' not in attrs and not new_class._meta.abstract:
+            for field in new_class._meta.fields:
+                if field.name == 'cmsplugin_ptr':
+                    app_label = new_class._meta.app_label
+                    model_name = new_class._meta.concrete_model._meta.model_name
```
Comment (czpython) [inline cms/models/pluginmodel.py]: huh? are you not already skipping non-concrete classes?

File: cms/models/pluginmodel.py [hunk 2]
```diff
@@ -48,6 +48,20 @@ def __new__(cls, name, bases, attrs):
         # create a new class (using the super-metaclass)
         new_class = super(PluginModelBase, cls).__new__(cls, name, bases, attrs)
 
+        # Set the one-to-one parent-child field to not create the reverse
+        # related descriptor to avoid name clashes for plugins with the same
+        # fieldnames. Skip this check if the parent-child field has been
+        # explicitly defined in the plugin subclass
+        #
+        # NOTE: We only do concrete classes, skipping the abstract ones
+        if 'cmsplugin_ptr' not in attrs and not new_class._meta.abstract:
```
Comment (czpython) [inline cms/models/pluginmodel.py]: we should test how this works with proxy models

File: cms/models/pluginmodel.py (hunk 1 above)
Comment (mkoistinen) [inline cms/models/pluginmodel.py]: Yes. strangely, _meta.model_name still comes back with the _deferred_path bit... This fixed it.

Comment (czpython) [inline cms/models/pluginmodel.py]: that's very strange.. I advise to look into why it happens.

File: cms/models/pluginmodel.py (hunk 2 above)
Comment (mkoistinen) [inline cms/models/pluginmodel.py]: I've yet to see a CMSPlugin that uses a proxy model... have you?

File: cms/models/pluginmodel.py (hunk 1 above)
Comment (yakky) [inline cms/models/pluginmodel.py]: @mkoistinen could you clarify more this point? What do you mean with ``..._deferred_path`` bit?

File: cms/models/pluginmodel.py (hunk 2 above)
Comment (yakky) [inline cms/models/pluginmodel.py]: I agree, we should explicitly skip proxied models too

**Comment by yakky** [comment]
Currently there is an issue with MTI: the plugins that inherit for a parent plugin overwrites the related_name, which must be kept to the parent one.
I'll send a PR with a test for this, and eventally a solution

**Comment by yakky** [comment]
@mkoistinen @czpython I opened a PR to your branch https://github.com/mkoistinen/django-cms/pull/2

**Comment by mkoistinen** [comment]
This is failing integration tests inside of the call to Django's `get_deleted_objects()`:

https://gist.github.com/mkoistinen/028dd9690d19f6ac1f86a6f791fa4545

Any ideas how to make this work?

**Comment by mkoistinen** [comment]
Django 1.8 and Python 3.4

**Pull request closed without merging**